x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/40_semantic_text_query.yml

```diff
@@ -90,6 +90,64 @@ setup:
   - close_to: { hits.hits.0._score: { value: 3.7837332e17, error: 1e10 } }
   - length: { hits.hits.0._source.inference_field.inference.chunks: 2 }
 
+---
+"Numeric query using a sparse embedding model":
+  - skip:
+      features: [ "headers", "close_to" ]
+
+  - do:
+      index:
+        index: test-sparse-index
+        id: doc_1
+        body:
+          inference_field: [40, 49.678]
+        refresh: true
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
+        Content-Type: application/json
+      search:
+        index: test-sparse-index
+        body:
+          query:
+            semantic:
+              field: "inference_field"
+              query: "40"
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc_1" }
+  - length: { hits.hits.0._source.inference_field.inference.chunks: 2 }
+
+---
+"Boolean query using a sparse embedding model":
+  - skip:
+      features: [ "headers", "close_to" ]
+
+  - do:
+      index:
+        index: test-sparse-index
+        id: doc_1
+        body:
+          inference_field: true
+        refresh: true
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
```

Comment:
Is content type needed here, as we're using booleans?

Reply:
This is a copy-paste from other YAML tests so that we can compare scores using the YAML assertions. Fun fact: If the test uses the `SMILE` format (which it will randomly do, unless you force JSON like is done here), then scores in search responses will be parsed as `float`, breaking the ability to check them using YAML assertions (which take `double` values).

We don't compare scores in this particular test, but it should be harmless to leave this so as to not create a landmine if we add score comparison in the future.

Reply:
Got it - thought it was referring to the actual values we were indexing instead of the score, as I saw no scores involved in the test.

 I'd say if it's not needed, don't add it - it confused me and probably will confuse others 🤷